File: activity/app/src/main/java/com/rikkei/training/activity/activity/EActivity.java
package com.rikkei.training.activity.activity;

import android.content.Intent;
import android.view.View;
import android.widget.Button;

import com.rikkei.training.activity.R;
import com.rikkei.training.activity.base.BaseActivity;

public class EActivity extends BaseActivity {
    private Button mButtonEC;

    @Override
    protected void initData() {

    }

    @Override
    protected void initViews() {
        mButtonEC = findViewById(R.id.btn_act_ec);
        mButtonEC.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(EActivity.this, CActivity.class);
                intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
                startActivity(intent);
            }
        });
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_e;
    }
}

File: Activitykotlin/app/src/main/java/com/rikkei/training/activitykotlin/Activity/EActivity.kt
package com.rikkei.training.activitykotlin.Activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.rikkei.training.activitykotlin.R
import kotlinx.android.synthetic.main.activity_e.*

class EActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_e)
        btn_gt_ec.setOnClickListener(View.OnClickListener {
            intent = Intent(this, CActivity::class.java)
            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
            startActivity(intent)
        })
    }
}
